service/src/test/java/com/bankbazaar/service/service/TripActivityServiceTest.java

```diff
@@ -0,0 +1,126 @@
+        tripDto.setTripName("name");
+        tripDto.setTripDescription("trip");
+        tripDto.setDestination("trip");
+        Date startDate = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripDto.setStartDate(startDate);
+        Date endDate = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripDto.setEndDate(endDate);
+        tripService.addTrip(tripDto, user.getUserId());
+    }
+
+    @Test
+    void addTripActivity() throws ParseException {
+        UserEntity user = userRepository.findByEmail("name@mail.com").get();
+        List<TripUserMapEntity> tripUserMap = tripUserMapRepository.findAll();
+        TripActivityDto tripActivityDto = new TripActivityDto();
+        tripActivityDto.setActivityStatus("PENDING");
+        tripActivityDto.setActivityDescription("trip activity");
+        Date date = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripActivityDto.setActivityTime(date);
+        tripActivityDto.setLocation("park");
+
+        service.addTripActivity(tripUserMap.get(0).getTripId(),tripActivityDto,user.getUserId());
+        assertEquals(tripActivityRepository.count(),1);
+    }
+
+    @Test
+    void findTripActivityById() throws ParseException {
+        UserEntity user = userRepository.findByEmail("name@mail.com").get();
+        List<TripUserMapEntity> tripUserMap = tripUserMapRepository.findAll();
+        TripActivityDto tripActivityDto = new TripActivityDto();
+        tripActivityDto.setActivityStatus("PENDING");
+        tripActivityDto.setActivityDescription("trip activity");
+        Date date = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripActivityDto.setActivityTime(date);
+        tripActivityDto.setLocation("park");
+
+        service.addTripActivity(tripUserMap.get(0).getTripId(),tripActivityDto,user.getUserId());
+
+        List<TripActivityDto> response = service.findTripActivityById(tripUserMap.get(0).getTripId(),user.getUserId());
+        assertEquals(response.size(),1);
+    }
+
+    @Test
+    void updateTripActivity() throws ParseException {
+        UserEntity user = userRepository.findByEmail("name@mail.com").get();
+        List<TripUserMapEntity> tripUserMap = tripUserMapRepository.findAll();
+        TripActivityDto tripActivityDto = new TripActivityDto();
+        tripActivityDto.setActivityStatus("PENDING");
+        tripActivityDto.setActivityDescription("trip activity");
+        Date date = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripActivityDto.setActivityTime(date);
+        tripActivityDto.setLocation("park");
+
+        TripActivityDto tripActivity = service.addTripActivity(tripUserMap.get(0).getTripId(),tripActivityDto,user.getUserId());
+        TripActivityDto response = service.updateTripActivity(tripActivity.getActivityId(),tripActivity.getTripId(),"APPROVED");
+
+        assertEquals(ActivityStatus.APPROVED.toString(), response.getActivityStatus());
```

Comment:
All this can be moved to a single test.
1. First validate that there are no activities for a trip.
2. Then add a couple of activities.
3. Validate find by tripId returns all
4. Then change status and validate
5. Finally delete an activity and verify that the no of activities returned in a trip has decreased